src/providers/details/tabs/Hosts/components/utils/onSaveHost.ts

```diff
@@ -0,0 +1,156 @@
+      model: SecretModel,
+      secretName,
+      secretNamespace,
+    });
+
+    await patchHost(host, ipAddress);
+    await patchSecret(secretData, encodedIpAddress, encodedUser, encodedPassword);
+  } else {
+    // Create a new host and secret pair
+
+    // Create a Secret
+    const secretData = {
+      apiVersion: 'v1',
+      data: {
+        ip: encodedIpAddress,
+        password: encodedPassword,
+        provider: encodedProvider,
+        user: encodedUser,
+      },
+      kind: 'Secret',
+      metadata: {
+        generateName: `${provider?.metadata?.name}-${inventory.id}-`,
+        labels: {
+          createdForResource: inventory.id,
+          createdForResourceType: 'hosts',
+        },
+        namespace: provider?.metadata?.namespace,
+      },
+      type: 'Opaque',
+    };
+    const createdSecret = await createHostSecret(secretData as IoK8sApiCoreV1Secret);
+
+    // Create Host
+    const newHostData = {
+      apiVersion: 'forklift.konveyor.io/v1beta1',
+      kind: 'Host',
+      metadata: {
+        name: `${provider?.metadata?.name}-${inventory.id}-config`,
+        namespace: provider?.metadata?.namespace,
+        ownerReferences: [
+          {
+            apiVersion: 'forklift.konveyor.io/v1beta1',
+            kind: 'Provider',
+            name: provider?.metadata?.name,
+            uid: provider?.metadata?.uid,
+          },
+        ],
+      },
+      spec: {
+        id: inventory.id,
+        ipAddress,
+        provider: {
+          name: provider?.metadata?.name,
+          namespace: provider?.metadata?.namespace,
+        },
+        secret: {
+          name: createdSecret?.metadata?.name,
+          namespace: createdSecret?.metadata?.namespace,
+        },
+      },
+    };
+    const createdHost = await createHost(newHostData as V1beta1Host);
+
+    // Patch Secret owner Ref
+    const ownerRef = {
+      name: createdHost?.metadata?.name,
+      uid: createdHost?.metadata?.uid,
+    };
+    await patchHostSecretOwner(createdSecret, ownerRef);
+  }
+};
+
+/**
+ * Saves hosts data, including associated secrets, for a given set of host+secret pairs.
+ * If a host already exists for a host pair, the host is updated with the new properties.
+ * Otherwise, a new host and secret are created.
+ */
+export const onSaveHost = async ({
+  hostPairs,
+  network,
+  passwd,
+  provider,
+  user,
+}: OnSaveHostParams) => {
+  const encodedUser = user ? Base64.encode(user) : undefined;
+  const encodedPassword = passwd ? Base64.encode(passwd) : undefined;
+  const encodedProvider = Base64.encode(provider?.metadata?.name ?? '');
+
+  const promises = hostPairs.map(async (hostPair) => {
+    // Look for the same network name in each host
+    const hostNetwork = hostPair?.inventory?.networkAdapters?.find(
+      ({ name }) => name === network.name,
+    );
+
+    if (!hostNetwork) {
+      throw new Error(
+        `can't find network ${network.name} on host ${hostPair?.host?.metadata?.name}`,
```

Comment:
Why throwing? Why crash the app?

Reply:
It won't crash the app since the ModalForm.onConfirm callback is handling exceptions by catching them and displaying them in modal
![Screenshot from 2025-05-20 18-38-36](https://github.com/user-attachments/assets/0da853f8-6640-4d08-bb07-6238ab6fb903)